crates/client/l1-messages/src/contract.rs

```diff
@@ -0,0 +1,49 @@
+use ethers::contract::abigen;
+use ethers::types::U256;
+use mp_felt::{Felt252Wrapper, Felt252WrapperError};
+use mp_transactions::HandleL1MessageTransaction;
+use starknet_api::transaction::Fee;
+
+use crate::error::L1MessagesWorkerError;
+
+abigen!(
+    L1Contract,
+    r"[
+	event LogMessageToL2(address indexed fromAddress, uint256 indexed toAddress, uint256 indexed selector, uint256[] payload, uint256 nonce, uint256 fee)
+]"
+);
+
+impl LogMessageToL2Filter {
+    pub fn try_get_fee(&self) -> Result<Fee, L1MessagesWorkerError> {
+        // Check against panic
+        // https://docs.rs/ethers/latest/ethers/types/struct.U256.html#method.as_u128
+        if self.fee > U256::from_big_endian(&(u128::MAX.to_be_bytes())) {
+            Err(L1MessagesWorkerError::ToFeeError)
+        } else {
+            Ok(Fee(self.fee.as_u128()))
+        }
+    }
+}
```

Comment:
I would not impl this on `LogMessageToL2Filter`. Just let the caller read the `pub` field and decide to do the conversion itself.

This is too specific. It only fulfills one specific use case you have (and you only have once) but it's not something that should be implemented on the type directly